### PR TITLE
Foundation for a Discord-facing module framework

### PR DESCRIPTION
**Pull request by SolarysDev**
- Create DiscordFacingManager. Used to match the given message to the module it's trying to invoke.
- Create IDiscordFacingModule. This is the main abstraction used by DiscordFacingManager to execute modules.
- Create DiscordFacingContext. Contains all the data needed for basic command execution. 
- Create DExampleModule. Self-Explanatory
- Modify Startup. Adding new modules to the service collection for use.

This framework should contain everything we need to create simple, Discord-invoked modules/commands if we ever need them. We can easily extend this if we need to, as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+appsettings.json
```
Comment (BillChirico) [inline .gitignore]: We don't want this file ignored. We'll keep out of repository values in user-secrets.

Comment (SolarysDev) [reply .gitignore]: Fixed

**Comment by SolarysDev** [comment]
.gitignore has been fixed to @BillChirico's specifications

**Comment by SolarysDev** [comment]
Implementing some suggestions from the Discord chat, will resubmit

**Comment by SolarysDev** [comment]
[DModule Framework](https://trello.com/c/G2gMESQI/1-dmodule-framework)

**Pull request closed without merging**